docs/channel-line-liff.md

```diff
@@ -0,0 +1,228 @@
+
+In this case, you need to add another `LINE_LIFF_ID` env to `.env` file, so your file will have at least those three LINE related environment variables:
+
+```
+LINE_ACCESS_TOKEN={your LINE access token from LINE Messaging API channel}
+LINE_CHANNEL_SECRET={your LINE channel secret from LINE Messaging API channel}
+LINE_LIFF_ID={your LIFF id from LINE Login channel}
+```
+
+You could find your LIFF ID in the LIFF URL. The format of LIFF URL looks like `line://app/{your LIFF ID}`.
+
+For more details about LIFF ID, please checkout [Registering LIFF App (in step 4)](https://developers.line.biz/en/docs/liff/registering-liff-apps/#registering-liff-app).
+
+### Adding Custom Routes to HTTP Server for LIFF Pages
+
+To serve LIFF webpages, we need to add additional routes to the server. Fortunately, [custom server](advanced-guides-custom-server#the-concept) come to the rescue!
+
+You could use express, koa, restify, or whatever you like, but we are going to use express in this guide. Before going down, make sure that you set up correctly according to [this guide](advanced-guides-custom-server#express).
+
+After having a custom server, you could add the following two routes into `server.js` to handle requests from LIFF.
+
+```js
+server.get('/send-id', (req, res) => {
+  res.json({ id: process.env.LINE_LIFF_ID });
+});
+
+server.get('/liff', (req, res) => {
+  const filename = path.join(`${__dirname}/liff.html`);
+  res.sendFile(filename);
+});
+```
+
+### Initializing the LIFF Page
+
+Before starting using any feature provided by LIFF, you need to create a `liff.html` file in the root directory of the project and copy the following code into it for LIFF initialization:
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>BOTTENDER LINE LIFF V2 DEMO</title>
+  </head>
+
+  <body>
+    <script src="https://static.line-scdn.net/liff/edge/2.1/sdk.js"></script>
+    <script>
+      function initializeLiff(myLiffId) {
+        liff
+          .init({
+            liffId: myLiffId,
+          })
+          .then(() => {
+            alert('LIFF init success!');
+          })
+          .catch(err => {
+            alert(`error: ${JSON.stringify(err)}`);
+          });
+      }
+
+      document.addEventListener('DOMContentLoaded', function() {
+        fetch(`/send-id`)
+          .then(function(reqResponse) {
+            return reqResponse.json();
+          })
+          .then(function(jsonResponse) {
+            let myLiffId = jsonResponse.id;
+            initializeLiff(myLiffId);
+          })
+          .catch(function(error) {
+            alert(`error: ${JSON.stringify(error)}`);
+          });
+      });
+    </script>
+  </body>
+</html>
+```
+
+It is worth mentioning that `liff.init()` and other `liff` methods all return `Promise` as results.
```

Comment:
This is not the truth. 
Only some methods would return `Promise`, not all methods.